provider/throttle_utils.py

```diff
@@ -141,3 +152,600 @@ def change_throttle_group(self, image, group_id):
+        Set expected result for testing. it stores in throttle["expected"].
+        The key-value pairs refer to default_expected.
+
+        :param expected: Dict of the expected result
+        :param reset: True for reset data before update.
+        """
+
+        if reset:
+            self._throttle["expected"] = copy.deepcopy(
+                ThrottleTester.raw_expected)
+        if expected:
+            for k, v in expected.items():
+                if isinstance(v, dict):
+                    self._throttle["expected"][k].update(expected[k])
+                else:
+                    self._throttle["expected"][k] = expected[k]
+
+    def set_fio_option(self, option):
+        """
+        Set the default fio option for all images without --filename
+
+        :param option: the fio option
+        """
+
+        self._fio_option = option
+
+    def build_default_option(self):
+        """
+        Generate default fio option for all images. It also generates expected
+        result according to throttle group property.
+        """
+
+        tgm = ThrottleGroupManager(self._vm)
+        attrs = tgm.get_throttle_group_props(self.group)
+
+        option = "--direct=1 --name=test --iodepth=1 --thread"
+        option += "  --output-format=json "
+
+        iops_size = attrs["iops-size"]
+        iops_size = 4096 if iops_size == 0 else iops_size
+
+        bps_read = attrs["bps-read"]
+        bps_read_max = attrs["bps-read-max"]
+        bps_read_max_length = attrs["bps-read-max-length"]
+        bps_total = attrs["bps-total"]
+        bps_total_max = attrs["bps-total-max"]
+        bps_total_max_length = attrs["bps-total-max-length"]
+        bps_write = attrs["bps-write"]
+        bps_write_max = attrs["bps-write-max"]
+        bps_write_max_length = attrs["bps-write-max-length"]
+        iops_read = attrs["iops-read"]
+        iops_read_max = attrs["iops-read-max"]
+        iops_read_max_length = attrs["iops-read-max-length"]
+        iops_total = attrs["iops-total"]
+        iops_total_max = attrs["iops-total-max"]
+        iops_total_max_length = attrs["iops-total-max-length"]
+        iops_write = attrs["iops-write"]
+        iops_write_max = attrs["iops-write-max"]
+        iops_write_max_length = attrs["iops-write-max-length"]
+
+        burst_read_iops = 0
+        burst_write_iops = 0
+        burst_total_iops = 0
+        normal_read_iops = 0
+        normal_write_iops = 0
+        normal_total_iops = 0
+
+        # reset expected result
+        self.set_throttle_expected(None, reset=1)
+
+        # count normal property
+        if iops_write != 0 or bps_write != 0:
+            bps = int(bps_write / iops_size)
+            iops = iops_write
+            if (iops >= bps != 0) or iops == 0:
+                normal_write_iops = bps
+            elif (bps >= iops != 0) or bps == 0:
+                normal_write_iops = iops
+            self.set_throttle_expected({"normal": {
+                "write": normal_write_iops}})
```

Comment:
We could wrap them into a function:
suggestion:
```
def _count_normal_iops_property(iops_type):
    bps = int(locals()['bps_%s' % iops_type] / iops_size)
    iops = locals()['iops_%s' % iops_type]
    if (iops >= bps != 0) or iops == 0:
        normal_iops = bps
    elif (bps >= iops != 0) or bps == 0:
        normal_iops = iops
    self.set_throttle_expected({"normal": {iops_type: normal_iops}})
    return normal_iops
```
```
if iops_write != 0 or bps_write != 0:
    normal_write_iops = _count_normal_iops_property('write')
```

Reply:
Thanks, but i think the solution do not meet the reqirement,please check the demo 

def test():
    a = 1
    b = 2

    def funca():
        print("global:", globals())
        globals()["a"] = 2
        globals()["b"] = 3

    def funcb():
        vars=locals()
        print("locals", vars)
        vars["a"] = 4
        vars["b"] = 5

    print(a, b)
    funcb()
    print(a, b)
    funca()
    print(a, b)


test()

result:
1 2
locals {}
1 2
global: {'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <_frozen_importlib_external.SourceFileLoader object at 0x7f1fc15f8d68>, '__spec__': None, '__annotations__': {}, '__builtins__': <module 'builtins' (built-in)>, '__file__': '/workdir/vbugs/py/t_locals.py', '__cached__': None, 'os': <module 'os' from '/usr/lib64/python3.6/os.py'>, 'tempfile': <module 'tempfile' from '/usr/lib64/python3.6/tempfile.py'>, 'test': <function test at 0x7f1fc1628ea0>}
1 2

Reply:
Yes, you can't change the item value of `locals()`, but you could get it.
From my side, There are no such requirements to set the value in your codes.
Just get it, like:
```
bps = int(bps_write / iops_size)
iops = iops_write
bps = int(bps_read / iops_size)
iops = iops_read
bps = int(bps_total / iops_size)
iops = iops_read
bps = int(bps_total / iops_size)
iops = iops_total
```
so I think it could meet your requirements.

Reply:
will update